### DISCUSSION TITLE
Help needed for Hashing Verification

### DISCUSSION THREAD

**Discussion by abrahimzaman360** [adonisjs/core#4720] [Help]
const user = await User.findBy('email', email)
      if (!user) {
        return response.status(404).send('User not found')
      }

      // check if password is correct
      if (await hash.verify(user.password, password)) {
        // generate token
        return;
      }
      
      everything works fine but it doesn't verify the database password hash with user input password!

**Comment by abrahimzaman360** [comment]
I don't understand what's the issue (:

**Comment by RomainLanz** [comment]
Hey @abrahimzaman360! 👋🏻

I'm not quite sure I fully understand the issue you're facing. Are you saying that `hash.verify(user.password, password)` is returning `false`?

It also looks like you're checking out the [Basic Example](https://docs.adonisjs.com/guides/authentication/verifying-user-credentials#basic-example) from our documentation on authentication.

Just a heads-up: this example isn’t the recommended approach for handling authentication.
![image](https://github.com/user-attachments/assets/00084f4c-222f-4b7f-a12c-18981b3d5e9a)

We probably need to work on making it clearer when code examples aren't ideal. 😄
